python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/io_manager.py

```diff
@@ -200,6 +154,133 @@ def bigquery_io_manager(init_context):
+                @staticmethod
+                def type_handlers() -> Sequence[DbTypeHandler]:
+                    return [BigQueryPandasTypeHandler()]
+
+            @asset(
+                key_prefix=["my_dataset"]  # my_dataset will be used as the dataset in BigQuery
+            )
+            def my_table() -> pd.DataFrame:  # the name of the asset will be the table name
+                ...
+
+            defs = Definitions(
+                assets=[my_table],
+                resources={
+                    "io_manager": MyBigQueryIOManager(project=EnvVar("GCP_PROJECT"))
+                }
+            )
+
+        You can tell Dagster in which dataset to create tables by setting the "dataset" configuration value.
+        If you do not provide a dataset as configuration to the I/O manager, Dagster will determine a dataset based
+        on the assets and ops using the I/O Manager. For assets, the dataset will be determined from the asset key,
+        as shown in the above example. The final prefix before the asset name will be used as the dataset. For example,
+        if the asset "my_table" had the key prefix ["gcp", "bigquery", "my_dataset"], the dataset "my_dataset" will be
+        used. For ops, the dataset can be specified by including a "schema" entry in output metadata. If "schema" is
+        not provided via config or on the asset/op, "public" will be used for the dataset.
+
+        .. code-block:: python
+
+            @op(
+                out={"my_table": Out(metadata={"schema": "my_dataset"})}
+            )
+            def make_my_table() -> pd.DataFrame:
+                # the returned value will be stored at my_dataset.my_table
+                ...
+
+        To only use specific columns of a table as input to a downstream op or asset, add the metadata "columns" to the
+        In or AssetIn.
+
+        .. code-block:: python
+
+            @asset(
+                ins={"my_table": AssetIn("my_table", metadata={"columns": ["a"]})}
+            )
+            def my_table_a(my_table: pd.DataFrame) -> pd.DataFrame:
+                # my_table will just contain the data from column "a"
+                ...
+
+        If you cannot upload a file to your Dagster deployment, or otherwise cannot
+        `authenticate with GCP <https://cloud.google.com/docs/authentication/provide-credentials-adc>`_
+        via a standard method, you can provide a service account key as the "gcp_credentials" configuration.
+        Dagster will store this key in a temporary file and set GOOGLE_APPLICATION_CREDENTIALS to point to the file.
+        After the run completes, the file will be deleted, and GOOGLE_APPLICATION_CREDENTIALS will be
+        unset. The key must be base64 encoded to avoid issues with newlines in the keys. You can retrieve
+        the base64 encoded with this shell command: cat $GOOGLE_APPLICATION_CREDENTIALS | base64
+    """
+
+    dataset: Optional[str] = Field(
+        None,
```

Comment:
minor nit: using `default=None` will make it more immediately obvious what is going on here.